### DISCUSSION TITLE
`tmp` folder on Vercel

### DISCUSSION THREAD

**Discussion by stoplion** [vercel/next.js#45566] [Help]
How do you set a `tmp` folder on Vercel (with NextJS?) to handle file uploads written to `tmp` folder.

I've seen multiple answers on how to do this, and they all have problems..

```const tmp = `os.tmpdir()/tmp```   
// ENOENT error on Vercel when I try to write to this location

`const tmp = `process.cwd()/tmp`   
// ENOENT error on Vercel when I try to write to this location

`const tmp = `/tmp`      
with some code that checks if the `tmp` dir exists, if not - create it..

```
try {
  await fs.readdir(tmp);
  console.log("Directory exists.");
} catch (err) {
  await fs.mkdir(tmp);
}
```

Results with this error:
```
read-only file system, mkdir '/var/task/tmp'
```

Also, in localhost `/tmp` saves the file to `/private/tmp` which I don't want. 

I just want the `/tmp` dir to BE IN THE PROJECT folder.

**Comment by stoplion** [comment]
this doc mentions https://vercel.com/guides/loading-static-file-nextjs-api-route
`const jsonDirectory = path.join(process.cwd(), 'someDir');` 

which results in `read-only file system` error

**Reply by br0wnD3v** [reply]
Any updates on this? Was trying to use the /tmp to keep some files before uploading them somewhere but fails 
```
[Error: EROFS: read-only file system, open 'tmp/to_upload']
```
